adapters/webhdfs-adapter/src/main/java/com/connexta/replication/adapters/webhdfs/WebHdfsNodeAdapter.java

```diff
@@ -0,0 +1,267 @@
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package com.connexta.replication.adapters.webhdfs;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.util.Map;
+import org.apache.http.Header;
+import org.apache.http.HttpEntity;
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ResponseHandler;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPut;
+import org.apache.http.client.methods.HttpRequestBase;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.entity.ContentType;
+import org.apache.http.entity.mime.MultipartEntityBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.codice.ditto.replication.api.NodeAdapter;
+import org.codice.ditto.replication.api.ReplicationException;
+import org.codice.ditto.replication.api.data.CreateRequest;
+import org.codice.ditto.replication.api.data.CreateStorageRequest;
+import org.codice.ditto.replication.api.data.DeleteRequest;
+import org.codice.ditto.replication.api.data.Metadata;
+import org.codice.ditto.replication.api.data.QueryRequest;
+import org.codice.ditto.replication.api.data.QueryResponse;
+import org.codice.ditto.replication.api.data.Resource;
+import org.codice.ditto.replication.api.data.ResourceRequest;
+import org.codice.ditto.replication.api.data.ResourceResponse;
+import org.codice.ditto.replication.api.data.UpdateRequest;
+import org.codice.ditto.replication.api.data.UpdateStorageRequest;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** Interacts with a remote Hadoop instance through the webHDFS REST API */
+public class WebHdfsNodeAdapter implements NodeAdapter {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(WebHdfsNodeAdapter.class);
+
+  private final URL webHdfsUrl;
+
+  private final CloseableHttpClient client;
+
+  /**
+   * Adapter to interact with a Hadoop instance through the webHDFS REST API
+   *
+   * @param webHdfsUrl the address of the REST API for the
+   * @param client performs the HTTP requests
+   */
+  public WebHdfsNodeAdapter(URL webHdfsUrl, CloseableHttpClient client) {
+    this.webHdfsUrl = webHdfsUrl;
+    this.client = client;
+  }
+
+  @Override
+  public boolean isAvailable() {
+    LOGGER.debug("Checking access to: {}", webHdfsUrl);
+
+    try {
+      URIBuilder builder = new URIBuilder(webHdfsUrl.toString());
+      builder.setParameter("op", "CHECKACCESS").setParameter("fsaction", "ALL");
```

Comment:
❗ Turns out the `fsaction` parameter doesn't work, at least for our configuration. I was able to verify in testing that having it results in a 400 status code, while removing it gives the correct 200.

Reply:
I removed the parameter, as suggested.